src/compiler/output-targets/test/output-lazy-loader.spec.ts

```diff
@@ -0,0 +1,73 @@
+import type * as d from '@stencil/core/declarations';
+import { mockBuildCtx, mockCompilerCtx, mockCompilerSystem, mockValidatedConfig } from '@stencil/core/testing';
+import { DIST, resolve } from '@utils';
+
+import { validateDist } from '../../config/outputs/validate-dist';
+import { outputLazyLoader } from '../output-lazy-loader';
+
+function setup(configOverrides: Partial<d.ValidatedConfig> = {}) {
+  const sys = mockCompilerSystem();
+  const config: d.ValidatedConfig = mockValidatedConfig({
+    ...configOverrides,
+    configPath: '/testing-path',
+    buildAppCore: true,
+    namespace: 'TestApp',
+    outputTargets: [
+      {
+        type: DIST,
+        dir: 'my-test-dir',
+      },
+    ],
+    srcDir: '/src',
+    sys,
+  });
+
+  config.outputTargets = validateDist(config, config.outputTargets);
+
+  const compilerCtx = mockCompilerCtx(config);
+  const writeFileSpy = jest.spyOn(compilerCtx.fs, 'writeFile');
+  const buildCtx = mockBuildCtx(config, compilerCtx);
+
+  return { config, compilerCtx, buildCtx, writeFileSpy };
+}
+
+describe('Lazy Loader Output Target', () => {
+  let config: d.ValidatedConfig;
+  let compilerCtx: d.CompilerCtx;
+  let writeFileSpy: jest.SpyInstance;
+
+  afterEach(() => {
+    writeFileSpy.mockRestore();
+  });
+
+  it('should write code for initializing polyfills when buildEs5=true', async () => {
+    ({ config, compilerCtx, writeFileSpy } = setup({ buildEs5: true }));
+    await outputLazyLoader(config, compilerCtx);
+
+    const expectedIndexOutput = `export * from '../esm/polyfills/index.js';
+export * from '../esm-es5/loader.js';`;
+    expect(writeFileSpy).toHaveBeenCalledWith(resolve('/my-test-dir/loader/index.js'), expectedIndexOutput);
```

Comment:
this is a little funny (the `resolve` call I mean) but I had to do this to get the tests passing on CI on Windows, you know, sometimes you gotta do what you gotta do

![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExMTc5b2hqM3kwYXp2Mjd2NHQyanM1dXo3djl6em1lYXJtaGdkZGZ0ayZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/PyLZ7uLdf96pi/giphy.gif)